app/Http/ViewHelpers/Employee/EmployeeShowViewHelper.php

```diff
@@ -893,4 +900,31 @@ public static function employeeCurrentAndPastPositions(Employee $employee, Compa
 
         return $positionCollection;
     }
+
+    /**
+     * Array containing information about the software associated with the
+     * employee.
+     *
+     * @param Employee $employee
+     * @param array $permissions
+     * @return Collection|null
+     */
+    public static function softwares(Employee $employee, array $permissions): ?Collection
+    {
+        if (! $permissions['can_see_software']) {
+            return null;
+        }
+
+        $softwares = $employee->softwares;
+
+        $softwareCollection = collect([]);
+        foreach ($softwares as $item) {
+            $softwareCollection->push([
+                'id' => $item->id,
+                'name' => $item->name,
+            ]);
+        }
+
+        return $softwareCollection;
```

Comment:
use laravel's collection map function:
```suggestion
        return $employee->softwares->map(function ($item) {
            return [
                'id' => $item->id,
                'name' => $item->name,
            ];
        });
```